### PR TITLE
FLEX-1998 Improves error feedback on catched exceptions.

### PR DESCRIPTION
**Pull request by Joep-Simons**

### CODE REVIEW

File: osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java [hunk 1]
```diff
@@ -286,14 +289,21 @@ public AddManufacturerResponse adddManufacturer(
...
                     new ValidationException(e.getConstraintViolations()));
+        } catch (final FunctionalException e) {
+            LOGGER.error("Exception adding manufacturer: {} ", e.getMessage(), e);
+            if (FunctionalExceptionType.EXISTING_MANUFACTURER == e.getExceptionType()) {
+                addManufacturerResponse.setResult(OsgpResultType.NOT_OK);
+                addManufacturerResponse.setDescription(ADD_MANUFACTURER_EXISTING_MANUFACTURER);
+                return addManufacturerResponse;
+            }
+            this.handleException(e);
         } catch (final Exception e) {
```
Comment (arnerippe) [inline osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java]: What happens with other FunctionalExceptionType exceptions than EXISTING_MANUFACTURER? Handled by handleException: OsgpException or TechnicalException is thrown. Is this the correct mechanism?

File: osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java [hunk 2]
```diff
@@ -465,6 +476,14 @@ public AddDeviceModelResponse addDeviceModel(@OrganisationIdentification final S
...
                     new ValidationException(e.getConstraintViolations()));
+        } catch (final FunctionalException e) {
+            LOGGER.error("Exception adding devicemodel: {} ", e.getMessage(), e);
+            if (FunctionalExceptionType.EXISTING_DEVICEMODEL == e.getExceptionType()) {
+                addDeviceModelResponse.setResult(OsgpResultType.NOT_OK);
+                addDeviceModelResponse.setDescription(ADD_DEVICEMODEL_EXISTING_DEVICEMODEL);
+                return addDeviceModelResponse;
+            }
+            this.handleException(e);
         } catch (final Exception e) {
```
Comment (arnerippe) [inline osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java]: What happens with other FunctionalExceptionType exceptions than EXISTING_DEVICEMODEL? Handled by handleException: OsgpException or TechnicalException is thrown. Is this the correct mechanism?

File: osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java [hunk 3]
```diff
@@ -604,13 +623,20 @@ public AddFirmwareResponse addFirmware(@OrganisationIdentification final String
...
                     new ValidationException(e.getConstraintViolations()));
+        } catch (final FunctionalException e) {
+            LOGGER.error("Exception adding firmware: {} ", e.getMessage(), e);
+            if (FunctionalExceptionType.EXISTING_FIRMWARE == e.getExceptionType()) {
+                addFirmwareResponse.setResult(OsgpResultType.NOT_OK);
+                addFirmwareResponse.setDescription(ADD_FIRMWARE_EXISTING_FIRMWARE);
+                return addFirmwareResponse;
+            }
+            this.handleException(e);
         } catch (final Exception e) {
```
Comment (arnerippe) [inline osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java]: What happens with other FunctionalExceptionType exceptions than EXISTING_FIRMWARE? Handled by handleException: OsgpException or TechnicalException is thrown. Is this the correct mechanism?

File: osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java (hunk 1 above)
Comment (Joep-Simons) [inline osgp-adapter-ws-core/src/main/java/com/alliander/osgp/adapter/ws/core/endpoints/FirmwareManagementEndpoint.java]: If FunctionalExceptio is not the exception I'd expect, the same action is being applied as a regular Exception.

**Pull request merged**